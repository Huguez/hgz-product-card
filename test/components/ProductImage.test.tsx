import React from "react"
import renderer from "react-test-renderer"
import { ProductCard, ProductImage } from '../../src/components';
import { product } from '../data/data';
// import img  from '../../src/assets/no-image.jpg'


describe( "ProductImage", () => {
   
   test( "Debe de mostrarse con la imagen personalizada", () =>{
      const wrapper = renderer.create( <ProductImage img="../../src/assets/no-image.jpg" /> )
      expect( wrapper.toJSON() ).toMatchSnapshot()
   } );

   test( "Debe de mostrar el componenete con la imagen del producto", () => {
      const wrapper = renderer.create( 
         <ProductCard product={ product } >
            {
               () => (
                  <ProductImage/>
               )
            }
         </ProductCard> 
      )
      
      expect( wrapper.toJSON() ).toMatchSnapshot()
   } );

} )
