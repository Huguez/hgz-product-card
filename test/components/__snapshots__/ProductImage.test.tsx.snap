// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Debe de mostrar el componenete con la imagen del producto 1`] = `
<div
  className="undefined  productCard "
>
  <img
    alt="Product image "
    className="undefined  productImg"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`ProductImage Debe de mostrarse con la imagen personalizada 1`] = `
<img
  alt="Product image "
  className="undefined  productImg"
  src="../../src/assets/no-image.jpg"
/>
`;
